foxpuppet/windows/browser/window.py

```diff
@@ -20,6 +20,10 @@ class BrowserWindow(BaseWindow):
     _file_menu_new_window_button_locator = (By.ID, "menu_newNavigator")
     _nav_bar_locator = (By.ID, "nav-bar")
     _notification_locator = (By.CSS_SELECTOR, "#notification-popup popupnotification")
+    _app_menu_notification_locator = (
+        By.CSS_SELECTOR,
+        "#appMenu-addon-installed-notification",
```

Comment:
This locator appears specific to the installed notification. Is this the only notification to appear within the app menu? Out of curiosity can you share a screenshot of what this looks like? If there are likely to be other notifications in this area, we should try to make this more generic.

Reply:
From my debugging there are other ones shown but not at the same time. Its the same area that shows the `Restart and Restore` for when firefox has an update. And they also seem to have different ID's.  That is also why I added  ```self.root.is_displayed()```.

Some examples: ```appMenu-update-restart-notification```, ```appMenu-update-manual-notification```.

[Screenshot](https://send.firefox.com/download/554609f68e/#vbXheWfPsKXHKtODy8ijHg)

Reply:
After doing a bit of investigation, it appears that we have a `<panel>` with id "appMenu-notification-popup". This has several child `<popupnotification>`s, only one of which is displayed. I would suggest that we could use the CSS locator `#appMenu-notification-popup popupnotification` and find the first one that is displayed. This would allow FoxPuppet to support the other app menu notifications in the future.

Reply:
Something like this should give you the first displayed notification:

```python
_app_menu_notification_locator = (By.CSS_SELECTOR, '#appMenu-notification-popup popupnotification')
notifications = self.selenium.find_elements(*_app_menu_notification_locator)
notification = next(n for n in notifications if n.is_displayed())
```

Reply:
I didn't put 2 and 2 together to realize that there are multiple elements with the same name... 😨